src/components/Common/Fields/hooks/useField.ts

```diff
@@ -0,0 +1,54 @@
+import type { RegisterOptions } from 'react-hook-form'
+import { useController, useFormContext } from 'react-hook-form'
+
+export type Transform<TEvent, TValue> = (event: TEvent) => TValue
+
+export interface UseFieldProps<TEvent, TValue = string> {
+  name: string
+  rules?: RegisterOptions
+  defaultValue?: TValue
+  errorMessage?: string
+  isRequired?: boolean
+  onChange?: (event: TEvent) => void
+  transform?: Transform<TEvent, TValue>
+}
+
+export function useField<TEvent, TValue = string>({
+  name,
+  rules = {},
+  defaultValue,
+  errorMessage,
+  isRequired = false,
+  onChange,
+  transform,
+}: UseFieldProps<TEvent, TValue>) {
+  const { control } = useFormContext()
+  const { field, fieldState } = useController({
+    name,
+    control,
+    rules: {
+      required: isRequired,
+      ...rules,
+    },
+    defaultValue,
+  })
+
+  const { ref, ...restFieldProps } = field
+
+  const handleChange = (event: TEvent) => {
+    const value = transform ? transform(event) : event
```

Comment:
This will allow us to do things like formatting the ssn and phone number without needing to manually set the value like we've been doing